### DISCUSSION TITLE
OpenTelemetry Metrics for outbox messages

### DISCUSSION THREAD

**Discussion by mathmotta** [BEagle1984/silverback#221] [Q&A] [answered]
Hello. Are there any default metrics that we can use/export? Would be nice to have a live feed of outbox messages or a counter of messages consumed by topics that we could visualize.

**Comment by BEagle1984** [comment, marked as answer]
When you say "outbox" do you mean produced messages in general or do you refer to the outbox pattern?

Regardless, the consumers expose a very minimal [StatusInfo](https://silverback-messaging.net/api/Silverback.Messaging.Broker.IConsumer.html#Silverback_Messaging_Broker_IConsumer_StatusInfo) that contains the count of consumed messages (per consumer, not per topic).

Furthermore, assuming that you are talking about Kafka, you also have the built-in statistics that can be consumed via the [callback](https://silverback-messaging.net/concepts/broker/kafka/kafka-events.html?tabs=offset-reset-startup) (they need to be enabled in the consumer configuration).

Finally, if you are talking about the outbox pattern, you can read some basic metrics like queue length and max age from the [IOutboxReader](https://silverback-messaging.net/api/Silverback.Messaging.Outbound.TransactionalOutbox.Repositories.IOutboxReader.html?q=IOutboxReader).

Silverback doesn't export these metrics but they are available for you to be collected and published however you want.
